File: backend/src/main/java/at/ac/tuwien/sepm/groupphase/backend/service/StatsService.java
package at.ac.tuwien.sepm.groupphase.backend.service;

import at.ac.tuwien.sepm.groupphase.backend.endpoint.dto.StatsDto;

public interface StatsService {

    /**
     * Provides total amount of events, tickets and news.
     *
     * @return StatsDto with the data
     */
    StatsDto getStats();
}

File: frontend/src/app/services/stats.service.ts
import { Injectable } from '@angular/core';
import {Globals} from '../global/globals';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';
import {Stats} from '../dtos/stats';

@Injectable({
  providedIn: 'root'
})
export class StatsService {

  private readonly baseUri: string = `${this.globals.backendUri}/stats`;

  constructor(private httpClient: HttpClient, private globals: Globals) {}

  /**
   * Get total amount of tickets, events and news
   */
  getStats(): Observable<Stats> {
    console.log(`Get stats.`);
    return this.httpClient.get<Stats>(this.baseUri);
  }
}
